src/tcp_address.cpp

```diff
@@ -129,7 +129,8 @@ static std::string make_address_string (const char *hbuf_,
     pos += hbuf_len;
     memcpy (pos, ipv6_suffix_, sizeof ipv6_suffix_ - 1);
     pos += sizeof ipv6_suffix_ - 1;
-    pos += sprintf (pos, "%d", ntohs (port_));
+    pos += snprintf (pos, max_port_str_length + 1 * sizeof (char), "%d",
```

Comment:
The [return value of `snprintf`](https://en.cppreference.com/w/c/io/fprintf) is the "number of characters (not including the terminating null character) which would have been written to buffer if `bufsz` was ignored, or a negative value if an encoding error (for string and character conversion specifiers) occurred". (Arguably this was a misdesign of `snprintf`.) So if there were a buffer overflow, `pos` would be calculated incorrectly and the call to the `std::string` constructor below would still be undefined behaviour.

A better way is to either use `sprintf_s` (requires `__STDC_WANT_LIB_EXT1__` to be defined before `#include <cstdio>`), or add an explicit assertion. I will file a PR shortly.